### PR TITLE
Adds anonymous users

### PR DESCRIPTION
**Pull request by hbcarlos**
I created the PR starting from #825.

Adds the anonymous users logic used in JupyterLab. It stores the user info in the cookie to persist between sessions as was done with the user id.

### CODE REVIEW

**Comment by vidartf** [comment]
#825 is now merged, so this can be rebased and continued now :)

**Comment by hbcarlos** [comment]
Hey, @vidartf thanks! This is ready for review.

**Comment by Zsailer** [comment]
Thanks, @hbcarlos! Great stuff here! 

Just a minor comment about "who" is responsible for choosing colors. I think this should depend on the frontend, not server.

File: jupyter_server/auth/utils.py [hunk 1]
```diff
@@ -74,3 +75,112 @@ def match_url_to_resource(url, regex_mapping=None):
...
+def get_anonymous_username() -> str:
+    """
+    Get a random user-name based on the moons of Jupyter.
+    This function returns names like "Anonymous Io" or "Anonymous Metis".
+    """
+    return moons_of_jupyter[random.randint(0, len(moons_of_jupyter) - 1)]
+
+
+# Using JupyterLab CSS variable because the colors may change with the theme
+user_colors = [
```
[the remark this answers is not in the document]
Comment (hbcarlos) [reply jupyter_server/auth/utils.py]: Thanks for looking at this @Zsailer.
I just removed the list of colors, I will keep them on the frontend.

**Review by Zsailer — APPROVED** [review]

**Comment by Zsailer** [comment]
The downstream tests in nbclassic is unrelated. Merging here. Thanks @hbcarlos!

**Comment by hbcarlos** [comment]
Thanks @Zsailer!

File: jupyter_server/auth/identity.py [hunk 2]
```diff
@@ -290,11 +292,28 @@ def user_to_cookie(self, user: User) -> str:
         Default is just the user's username.
         """
         # default: username is enough
```
Comment (vidartf) [inline jupyter_server/auth/identity.py:294]: This comment and the docstring are now out of date :) I think also we should clarify whether this cookie format is now a required, de-facto standard by JupyterLab (will lab stop working if this cookie structure is broken?).

**Pull request merged**